src/RpcServer/RpcServer.cs

```diff
@@ -52,9 +52,13 @@ private bool CheckAuth(HttpContext context)
         {
             if (string.IsNullOrEmpty(settings.RpcUser)) return true;
 
-            context.Response.Headers["WWW-Authenticate"] = "Basic realm=\"Restricted\"";
-
             string reqauth = context.Request.Headers["Authorization"];
+            if (string.IsNullOrEmpty(reqauth))
+            {
+                context.Response.Headers["WWW-Authenticate"] = "Basic realm=\"Restricted\"";
+                context.Response.StatusCode = 401;
```

Comment:
```suggestion
                context.Response.StatusCode = 401;
                return false;
```

Reply:
if you do that then window won't popup in the browser. Its not valid spec for using WWW-Authenticate header you need to return 401 or remove www-authenticate header from the response.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#digest_authentication_with_sha-256_and_md5
>The first time the client requests the document, no [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header field is sent. Here the server responds with an HTTP 401 message that includes a challenge for each digest algorithm it supports, in its order of preference (SHA256 and then MD5)

Also see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
>The server responds with a [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) Unauthorized message that includes at least one [WWW-Authenticate](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate) header.

Reply:
It will return false later

Reply:
yes, But the server with response with Status Code 200 (OK). Which is invalid. If you are using those headers and creds are wrong or not supplied.